src/primitives/compression.cpp

```diff
@@ -0,0 +1,358 @@
+            int length = m_signature.size()-1;
+            secp256k1_ecdsa_signature sig;
+            if (secp256k1_ecdsa_signature_parse_der(ctx, &sig, &m_signature[0], length)) {
+                if (secp256k1_ecdsa_signature_serialize_compact(ctx, &m_signature[0], &sig)) {
+                    m_signature.resize(64);
+                    m_compressedSignature = true;
+                }
+            }
+        } else if (txin.scriptWitness.stack.size() && scriptPubKey.IsPayToTaproot()) {
+            if (txin.scriptWitness.stack.at(0).size() != 64) {
+                m_signature = txin.scriptWitness.stack.at(0);
+                m_hashType = m_signature[m_signature.size()-1];
+                m_signature.pop_back();
+            }
+            m_compressedSignature = true;
+        }
+        if (!m_compressedSignature) {
+            CDataStream stream(SER_DISK, 0);
+            stream << VARINT(txin.scriptSig.size());
+            if (txin.scriptSig.size())
+                stream.write(MakeByteSpan(txin.scriptSig));
+            stream << VARINT(txin.scriptWitness.stack.size());
+            for (size_t index = 0; index < txin.scriptWitness.stack.size(); index++) {
+                stream << VARINT(txin.scriptWitness.stack.at(index).size());
+                stream.write(MakeByteSpan(txin.scriptWitness.stack.at(index)));
+            }
+            m_signature.resize(stream.size());
+            stream.read(MakeWritableByteSpan(m_signature));
+        }
+    }
+    m_nSequence = txin.nSequence;
+}
+
+std::string CCompressedTxIn::ToString() const
+{
+    return strprintf("CCompressedTxIn(prevout=%s, signature=%s, compressed=%b, hashType=%u, nSequence=%u)",
+    m_prevout.ToString(),
+    HexStr(m_signature),
+    m_compressedSignature,
+    m_hashType,
+    m_nSequence);
+}
+
+CCompressedTxOut::CCompressedTxOut(const CTxOut& txout) {
+    m_nValue = txout.nValue;
+    std::vector<std::vector<unsigned char>> solutions;
+    m_scriptType = Solver(txout.scriptPubKey, solutions);
+    switch (m_scriptType) {
+        case TxoutType::PUBKEY:
+        case TxoutType::PUBKEYHASH:
+        case TxoutType::SCRIPTHASH:
+        case TxoutType::WITNESS_V0_KEYHASH:
+        case TxoutType::WITNESS_V0_SCRIPTHASH:
+        case TxoutType::WITNESS_V1_TAPROOT:
+            m_scriptPubKey = solutions.at(0);
+            break;
+        default:
+            m_scriptPubKey.resize(txout.scriptPubKey.size());
+            copy(txout.scriptPubKey.begin(), txout.scriptPubKey.end(), m_scriptPubKey.begin());
+    }
+}
+
+std::string CCompressedTxOut::ToString() const
+{
+    return strprintf("CCompressedTxOut(scriptPubKey=%s, scriptType=%s, nValue=%u)",
+    HexStr(m_scriptPubKey),
+    GetTxnOutputType(m_scriptType),
+    m_nValue);
+}
+
+CCompressedTransaction::CCompressedTransaction(secp256k1_context* ctx, const CTransaction tx, const std::vector<CCompressedTxId>& txids, const std::vector<CScript>& scriptPubKeys) {
```

Comment:
In 89b52782f8d6aca75d49048da1e137fbf615c0e6

`TxId` should be spelled `Txid` everywhere.

Reply:
Solved, Thanks!